src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore
+{
+    /// <summary>
+    /// A <see cref="IFileProvider"/> for serving static web assets during development.
+    /// </summary>
+    internal class StaticWebAssetsFileProvider : IFileProvider
+    {
+        private static readonly StringComparison FileSystemBasePathComparisonMode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+            StringComparison.OrdinalIgnoreCase :
+            StringComparison.Ordinal;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="StaticWebAssetsFileProvider"/>.
+        /// </summary>
+        /// <param name="pathPrefix">The path prefix under which the files in the <paramref name="contentRoot"/> folder will
+        /// be mapped.</param>
+        /// <param name="contentRoot">The absolute path to the content root associated with the static web assets.</param>
+        public StaticWebAssetsFileProvider(string pathPrefix, string contentRoot)
+        {
+            BasePath = pathPrefix.StartsWith("/") ? pathPrefix : "/" + pathPrefix;
+            InnerProvider = new PhysicalFileProvider(contentRoot);
+        }
```

Comment:
I think this deserves some comments about why we need to write a new file provider. We could also consider adding support for *base paths* to `PhysicalFileProvider` which would result in this being more efficient.

Reply:
This is very specialized implementation for this scenario so I wouldn't pollute PhysicalFileProvider with this concept, if the need arises, we can always integrate it at a later point.

Reply:
Also, regarding efficiency, this is for development scenarios only (it doesn't affect published apps) so i'm not super concerned about the performance.

Even if you plug this in, static files will be considered first, so it is unlikely you see a hit here.